flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/AvailableNodesGenerator.java

```diff
@@ -230,24 +231,54 @@ private static List<Item> fetchConnections(ModuleID moduleId, String parentSymbo
         for (FunctionResult connectorAction : connectorActions) {
-            NodeBuilder actionBuilder = NodeBuilder.getNodeFromKind(NodeKind.ACTION_CALL);
-            actionBuilder
-                    .metadata()
-                        .label(connectorAction.name())
-                        .icon(CommonUtils.generateIcon(connector.org(), connector.packageName(), connector.version()))
-                        .description(connectorAction.description())
-                        .stepOut()
-                    .codedata()
-                        .node(NodeKind.ACTION_CALL)
-                        .org(connector.org())
-                        .module(connector.packageName())
-                        .object(NewConnection.CLIENT_SYMBOL)
-                        .symbol(connectorAction.name())
-                        .parentSymbol(parentSymbol)
-                        .id(connectorAction.functionId());
+            NodeBuilder actionBuilder = connectorAction.kind() == Function.Kind.RESOURCE ?
+                    getResourceActionNode(connectorAction, connector, parentSymbol)
+                    : getActionNode(connectorAction, connector, parentSymbol);
             availableNodes.add(actionBuilder.buildAvailableNode());
         }
         return availableNodes;
     }
 
+    private static NodeBuilder getActionNode(FunctionResult connectorAction, FunctionResult connector,
+                                             String parentSymbol) {
+        NodeBuilder actionBuilder = NodeBuilder.getNodeFromKind(NodeKind.ACTION_CALL);
+        actionBuilder
+                .metadata()
+                    .label(connectorAction.name())
+                    .icon(CommonUtils.generateIcon(connector.org(), connector.packageName(), connector.version()))
+                    .description(connectorAction.description())
+                    .stepOut()
+                .codedata()
+                    .node(NodeKind.ACTION_CALL)
+                    .org(connector.org())
+                    .module(connector.packageName())
+                    .object(NewConnection.CLIENT_SYMBOL)
+                    .symbol(connectorAction.name())
+                    .parentSymbol(parentSymbol)
+                    .id(connectorAction.functionId());
+        return actionBuilder;
+    }
+
+    private static NodeBuilder getResourceActionNode(FunctionResult connectorAction, FunctionResult connector,
+                                                     String parentSymbol) {
+        NodeBuilder actionBuilder = NodeBuilder.getNodeFromKind(NodeKind.RESOURCE_ACTION_CALL);
+        actionBuilder
+                .metadata()
+                    .label(connectorAction.name() + ":" + connectorAction.resourcePath())
+                    .icon(CommonUtils.generateIcon(connector.org(), connector.packageName(), connector.version()))
+                    .description(connectorAction.description())
+                    .keywords(List.of(Function.Kind.RESOURCE.name()))
```

Comment:
I am not sure if this is the correct field for adding this, as this is used to capture the keywords specified in the Ballerina.toml. Better if we can bring a separate field to capture the Function.Kind.

Reply:
fixed